### PR TITLE
`ProxyPNL`, `historic_portfolio_vol()` - full/inf. lookback window

### PR DESCRIPTION
**Pull request by Magnus167**

### CODE REVIEW

File: macrosynergy/pnl/historic_portfolio_volatility.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def _weighted_covariance(
 
     xnans, ynans = np.isnan(x), np.isnan(y)
     wmask = xnans | ynans
-    weightslen = min(sum(~wmask), lback_periods)
+    weightslen = min(sum(~wmask), lback_periods if lback_periods > 0 else len(x))
```
Comment (lsimonsen) [inline macrosynergy/pnl/historic_portfolio_volatility.py:63]: Do we need to be explicit about only -1 is full look back? So basically:

```python
assert lback_periods == -1 or lback_periods > 0
```

Just to insure the user don't accidentally put in -21 days or similar?

File: macrosynergy/pnl/historic_portfolio_volatility.py [hunk 2]
```diff
@@ -125,23 +125,32 @@ def _mask_nans(
     remove_zeros: bool,
 ) -> pd.DataFrame:
     """Mask NaNs in the dataframe"""
+    lback_periods = 0 if lback_periods < 0 else lback_periods
```
Comment (lsimonsen) [inline macrosynergy/pnl/historic_portfolio_volatility.py:128]: Does this work with our `lback_periods = -1` for full sample or is the logic that 0 or less is full sample?

Comment (Magnus167) [reply macrosynergy/pnl/historic_portfolio_volatility.py:128]: No, the only acceptable negative value is -1.

https://github.com/macrosynergy/macrosynergy/blob/d604e10c97a61d33744e86ba6611ef8e783a0482/macrosynergy/pnl/historic_portfolio_volatility.py#L205

File: macrosynergy/pnl/historic_portfolio_volatility.py (hunk 1 above)
Comment (Magnus167) [reply macrosynergy/pnl/historic_portfolio_volatility.py:63]: This is asserted earlier at https://github.com/macrosynergy/macrosynergy/blob/d604e10c97a61d33744e86ba6611ef8e783a0482/macrosynergy/pnl/historic_portfolio_volatility.py#L205

**Pull request merged**